src/zenml/integrations/kserve/custom_deployer/zenml_custom_model.py

```diff
@@ -0,0 +1,170 @@
+        """
+        super().__init__(model_name)
+        self.name = model_name
+        self.model_uri = model_uri
+        self.predict_func = import_class_by_path(predict_func)
+        self.model = None
+        self.ready = False
+
+    def load(self) -> bool:
+        """Load the model.
+
+        This function loads the model into memory and sets the ready flag to True.
+
+        The model is loaded using the materializer, by saving the information of
+        the artifact to a YAML file in the same path as the model artifacts at
+        the preparing time and loading it again at the prediction time by
+        the materializer.
+
+        Returns:
+            True if the model was loaded successfully, False otherwise.
+
+        """
+        try:
+            from zenml.utils.materializer_utils import load_model_from_metadata
+
+            self.model = load_model_from_metadata(self.model_uri)
+        except Exception as e:
+            logger.error("Failed to load model: {}".format(e))
+            return False
+        self.ready = True
+        return self.ready
+
+    def predict(self, request: Dict[str, Any]) -> Dict[str, Any]:
+        """Predict the given request.
+
+        The main predict function of the model. This function is called by the
+        KServe server when a request is received. Then inside this function,
+        the user-defined predict function is called.
+
+        Args:
+            request: The request to predict in a dictionary. e.g. {"instances": []}
+
+        Returns:
+            The prediction dictionary.
+
+        Raises:
+            Exception: If function could not be called.
+            NotImplementedError: If the model is not ready.
+        """
+        if self.predict_func is not None:
+            try:
+                prediction = {
+                    "predictions": self.predict_func(
+                        self.model, request["instances"]
+                    )
+                }
+            except Exception as e:
+                raise Exception("Failed to predict: {}".format(e))
+            if isinstance(prediction, dict):
+                return prediction
+            else:
+                raise Exception("Prediction is not a dictionary.")
+        else:
+            raise NotImplementedError("Predict function is not implemented")
+
+
+@click.command()
+@click.option(
+    "--model_uri",
+    default=DEFAULT_LOCAL_MODEL_DIR,
+    type=click.STRING,
+    help="The directory where the model is stored locally.",
+)
+@click.option(
+    "--model_name",
+    default=DEFAULT_MODEL_NAME,
+    required=True,
+    type=click.STRING,
+    help="The name of the model to deploy. This important for the Kserve server.",
+)
+@click.option(
+    "--predict_func",
+    required=True,
+    type=click.STRING,
+    help="The path to the custom predict function defined by the user.",
+)
+def main(model_name: str, model_uri: str, predict_func: str) -> None:
```

Comment:
The way the custom deployment server works with the KServe server is by implementing a custom model class and passing it to the KServe server and then starting the server. Because custom classes usually need some parameters to be passed to the model, the parameters are passed from the entry point to the main function as arguments and then passed to the model class constructor.